### PR TITLE
Convert Header tests to TS

### PR DESCRIPTION
**Pull request by sebastien-comeau**
### Description

List of proposed changes:

- Convert Header component tests to TS

### What to test for/How to test

Load the branch code locally and run `npm run test:unit`

### CODE REVIEW

File: __tests__/components/Header.test.tsx [hunk 1]
```diff
@@ -1,50 +1,43 @@
...
-    render(<Header />)
+    render(<Header gocLink="testGocLink" skipToMainText="testSkipToMainText" />)
     const HeaderLang = screen.getByText('Français')
     expect(HeaderLang).toBeInTheDocument()
   })
 
   it('renders Header in French', () => {
-    useRouter.mockImplementation(() => ({ ...defaultRouterObj, locale: 'fr' }))
-    render(<Header />)
+    ;(useRouter as jest.Mock).mockImplementationOnce(() => ({
```
Comment (krischarbonneau) [inline __tests__/components/Header.test.tsx:28]: Stray semicolon here?

**Review by krischarbonneau — COMMENTED** [review]
Looks, good, just one small comment

File: __tests__/components/Header.test.tsx (hunk 1 above)
Comment (sebastien-comeau) [reply __tests__/components/Header.test.tsx:28]: Yeah, it's added when I save. I believe Prettier is adding it.
https://prettier.io/docs/en/rationale.html#semicolons

Comment (sebastien-comeau) [reply __tests__/components/Header.test.tsx:28]: https://standardjs.com/rules.html#semicolons

Comment (sebastien-comeau) [reply __tests__/components/Header.test.tsx:28]: I can change it like this if you prefer:
```tsx
const mockUseRoutrer = (useRouter as jest.Mock)
mockUseRoutrer.mockImplementationOnce(...)
```

Comment (sebastien-comeau) [reply __tests__/components/Header.test.tsx]: Change applied. TIL semicolon usage with no-semicolon javascript. hehe

Comment (krischarbonneau) [reply __tests__/components/Header.test.tsx]: TIL as well haha

**Review by krischarbonneau — APPROVED** [review]
Looks good!

**Pull request merged**